reason_4.0/lib/core/classes/entity.php

```diff
@@ -237,17 +405,50 @@ function get_value_refresh( $col ) // {{{
 	{
 		$this->refresh_values();
 		if( empty( $this->_values ) )
+		{
 			$this->_values = get_entity_by_id( $this->_id );
+			$this->full_fetch_performed(true);
+		}
 		if( !empty( $this->_values[ $col ]) OR (isset($this->_values[$col]) AND strlen($this->_values[ $col ]) > 0) )
 			return $this->_values[ $col ];
 		elseif(!array_key_exists($col, $this->_values))
 		{
-			trigger_error('"'.$col.'" field not retrieved from database');
+			$val = $this->get_value_from_delegate( $col );
+			if(null !== $val)
+				return $val;
+			else
+				trigger_error('"'.$col.'" field not retrieved from database');
 			// echo '<pre>';debug_print_backtrace();echo '</pre>';
 		}
 		return false;
 	} // }}}
 	
+	/**
+	 * Get a value from delegate
+	 * @param string $col the column/key requested
+	 * @param mixed $delegate_path the delegate path or null for first delegate that
+	 * supports this value
+	 * @return mixed NULL if no delegate supports, mixed non-null value otherwise
+	 */
+	function get_value_from_delegate( $col, $delegate_path = null )
+	{
+		$delegates = $this->get_delegates();
+		if(!$delegate_path)
+		{
+			foreach($delegates as $delegate)
+			{
+				$val = $delegate->get_value( $col );
+				if(NULL !== $val)
+					return $val;
+			}
+		}
+		elseif(isset($delegates[$delegate_path]))
+		{
+			return $delegate->get_value( $col );
```

Comment:
I think `$delegate` will not be defined in this code path. I don't see this path exercised in this PR.